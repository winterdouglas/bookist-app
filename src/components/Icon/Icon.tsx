import { IconProps as RNVIconProps } from "react-native-vector-icons/Icon";
import RNVIcon from "react-native-vector-icons/Ionicons";
import Ionicons from "react-native-vector-icons/dist/glyphmaps/Ionicons.json";

export type IconName = keyof typeof Ionicons;

export type IconProps = RNVIconProps & {
  name: IconName;
};

export const Icon = ({ name, ...props }: IconProps) => {
  return <RNVIcon name={name} {...props} />;
};
