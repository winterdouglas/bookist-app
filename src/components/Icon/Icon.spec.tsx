import renderer from "react-test-renderer";
import { Icon } from "./Icon";

it("renders correctly", () => {
  const tree = renderer.create(<Icon name="alarm" />).toJSON();
  expect(tree).toMatchSnapshot();
});
