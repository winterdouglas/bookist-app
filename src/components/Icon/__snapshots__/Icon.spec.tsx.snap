// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
>
  Ó®ç
</Text>
`;
